### DISCUSSION TITLE
Changing password complexity

### DISCUSSION THREAD

**Discussion by curelom** [oqtane/oqtane.framework#1327] [Q&A]
How can we change password complexity in oqtane?

**Comment by hishamco** [comment]
Simply you can change the `IdentityOptions`

https://github.com/oqtane/oqtane.framework/blob/75c336454b00effbed8e188a5d997d4742cf05f2/Oqtane.Server/Startup.cs#L139-L144

**Reply by curelom** [reply]
Thank you.  I was afraid I was going to have to change the framework code, but if there is no other way, that's what I'll do.

**Reply by hishamco** [reply]
If you have a custom server startup class you can reconfigure the `IdentityOptions`, otherwise you should need to touch these settings

I think we could expose such settings in `appsettings.json` to make it easier to configure

**Reply by BGillham** [reply]
I cannot find a Startup.cs in my installation... Can someone please advise further... Sorry if the question seems inane but I am brand-new to Oqtane (am a DNN guy)...

**Reply by leigh-pointer** [reply]
Hi @BGillham 
pictures speak a 1000 words.

![image](https://user-images.githubusercontent.com/9447612/117665259-ecc3b800-b1a2-11eb-9c9d-8e3fdb5c228d.png)

You should find it a the root.

**Reply by leigh-pointer** [reply]
@BGillham If you want to access Startup from your module you will need to add the class to your project.  It is not there by default. You will need to implement the interfaces.

**Reply by sbwalker** [reply]
I will add this enhancement to the roadmap for the 2.2 release where we intend to broaden the authentication provider capabilities

**Reply by mixlplix** [reply]
I agree it would be nice to be able to configure identity options without having to change the startup.cs
